pkg/logger/handler.go

```diff
@@ -125,7 +125,11 @@ func (eh *LoggerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	} else {
 		eh.log.Info("Failed to proxy request", "status code", rr.Code)
 	}
-
+	contentType := rr.Header().Get("Content-Type")
```

Comment:
we can move this before line 109 and replace the condition:
```
	if rr.Code == http.StatusOK {
```
with
```
	if contentType == "application/json" {
```
By doing so, we'll actually write the error log in the cloudevent as well, as long as it's json, as per the request of @javierdlrm

Reply:
It looks good! Just another remark, having a look here https://github.com/theofpa/kfserving/commit/fa0caf60f0d228769993903bc6049c905902aa48, in case the response is OK but `Content-Type: application/json` is not present in the headers, `Failed to proxy request` will appear in the logs although the request was successful. In that case, the log is misleading.

Reply:
Maybe we can move line 127 ([commit](https://github.com/theofpa/kfserving/commit/fa0caf60f0d228769993903bc6049c905902aa48)) into a different if-condition checking for OK status code, just below the content-type condition.